activitypub-core/src/main/java/org/xwiki/contrib/activitypub/internal/json/ActivityPubObjectDeserializer.java

```diff
@@ -57,32 +59,40 @@
      */
     @Override
     public ActivityPubObject deserialize(JsonParser jsonParser, DeserializationContext deserializationContext)
-        throws IOException, JsonProcessingException
+        throws IOException
     {
         ObjectMapper mapper = (ObjectMapper) jsonParser.getCodec();
         ObjectNode root = mapper.readTree(jsonParser);
 
+        Logger logger = (Logger) deserializationContext.findInjectableValue(LOGGER_KEY, null, null);
+
         JsonNode type = root.get("type");
-        Class<? extends ActivityPubObject> instanceClass;
+        Optional<Class<? extends ActivityPubObject>> instanceClass;
         if (type != null) {
-            instanceClass = findClass(type.asText());
+            instanceClass = findClass(type.asText(), logger);
+        } else {
+            instanceClass = Optional.of(ActivityPubObject.class);
+        }
+        if (instanceClass.isPresent()) {
+            return mapper.treeToValue(root, instanceClass.get());
         } else {
-            instanceClass = ActivityPubObject.class;
+            return null;
         }
-        return mapper.treeToValue(root, instanceClass);
     }
 
-    private Class<? extends ActivityPubObject> findClass(String type)
+    private Optional<Class<? extends ActivityPubObject>> findClass(String type, Logger logger)
     {
         Class<? extends ActivityPubObject> classInPackage;
         for (String packageExtension : PACKAGES_ENTITIES) {
             classInPackage = findClassInPackage(packageExtension, type);
             if (classInPackage != null) {
-                return classInPackage;
+                return Optional.of(classInPackage);
             }
         }
-        // TODO: validate that this fail fast strategy is relevant.
-        throw new RuntimeException(MessageFormat.format("Type [{0}] not found", type));
+        
+        logger.debug("ActivityPub Object type [{}] not found.", type);
```

Comment:
I'm not sure I'd have logged a simple debug. We were failing directly the answer before so we were considering this as pretty important. IMO we should at least log an info, and I'd probably go for a warning.

Reply:
Log level updated to warn. We can consider lowering it to debug or info once the application gain in stability.